keel-slack/src/main/kotlin/com/netflix/spinnaker/keel/slack/callbacks/ManualJudgmentCallbackHandler.kt

```diff
@@ -0,0 +1,132 @@
+import com.slack.api.model.kotlin_extension.block.withBlocks
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Component
+import java.time.Clock
+import java.time.Instant
+
+/**
+ * This handler will handel slack callbacks coming from Manual Judgment notifications.
+ * First, it will update the constraint status based on the user response (either approve/reject)
+ * Second, it will construct an updated notification with the action preformed and the user how did it.
+ */
+@Component
+class ManualJudgmentCallbackHandler(
+  private val clock: Clock,
+  private val repository: KeelRepository,
+  private val slackService: SlackService
+) {
+
+  private val log by lazy { LoggerFactory.getLogger(javaClass) }
+
+  //Updating the constraint status based on the user response (either approve/reject)
+  fun handleMJResponse(slackCallbackResponse: SlackCallbackHandler.SlackCallbackResponse) {
+    val constraintUid = slackCallbackResponse.constraintId
+    val currentState = repository.getConstraintStateById(parseUID(constraintUid))
+      ?: throw SystemException("constraint@callbackId=$constraintUid", "constraint not found")
+
+    val user = slackService.getEmailByUserId(slackCallbackResponse.user.id)
+    val actionStatus = slackCallbackResponse.actions.first().value
+
+    log.debug(
+      "Updating constraint status based on notification interaction: " +
+        "user = $user, status = $actionStatus}"
+    )
+
+    repository
+      .storeConstraintState(
+        currentState.copy(
+          status = ConstraintStatus.valueOf(actionStatus),
+          judgedAt = Instant.now(),
+          judgedBy = user
+        )
+      )
+
+    //convert status to actual action --> like OVERRIDE_PASS to "approve"
+    val actionPerformed = actionsMap[actionStatus]
+    if (actionPerformed == null) {
+      log.warn("can't map slack action status to an actual action and therefore can't send an updated notification.")
+      return
+    }
+
+    respondToCallback(slackCallbackResponse, actionPerformed)
+  }
+
+  fun respondToCallback(slackCallbackResponse: SlackCallbackHandler.SlackCallbackResponse, actionPerformed: String) {
+    log.debug("Responding to Slack callback via ${slackCallbackResponse.response_url}")
+
+    val fallbackText = "@${slackCallbackResponse.user.name} hit " +
+      "$actionPerformed on <!date^${clock.instant().epochSecond}^{date_num} {time_secs}|fallback-text-include-PST>"
+
+    //construct an update notification
+    val updatedNotification = updateMJNotification(
+      slackCallbackResponse.message["blocks"] as List<Any>,
+      fallbackText,
+      actionPerformed)
+
+    //send notification using the response url slack provides with the original message
+    slackService.respondToCallback(
+      slackCallbackResponse.response_url,
+      updatedNotification,
+      fallbackText)
+  }
+
+  fun updateMJNotification(originalBlocks: List<Any>, fallbackText: String, action: String): List<LayoutBlock> {
```

Comment:
Consider renaming for clarity:
```suggestion
  fun updateManualJudgementNotification(originalBlocks: List<Any>, fallbackText: String, action: String): List<LayoutBlock> {
```